### PR TITLE
[CloudService] Init for CloudService

### PR DESCRIPTION
**Pull request by wyunchi-ms**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@wyunchi-ms  Please fix the credential scan error

File: src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1 [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        if (-not $PSBoundParameters.ContainsKey("SubscriptionId")) 
+        {
+            $SubscriptionId = (Get-AzContext).Subscription.Id
+        }
+
+        if ($PSBoundParameters.ContainsKey("CloudService"))
+        {
+            $elements = $CloudService.Id.Split("/")
+            $SubscriptionId = $elements[2]
```
Comment (msJinLei) [inline src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1:58]: Please check subscript out of bound

File: src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1 [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            $ResourceGroupName = $elements[4]
+            $CloudServiceName = $CloudService.Name
```
Comment (msJinLei) [inline src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1:60]: what to happen if $CloudService.Name is $null?

File: src/CloudService/custom/Get-AzCloudServicePublicIPAddress.ps1 [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    process {
+
+        if (-not $PSBoundParameters.ContainsKey("SubscriptionId")) 
+        {
+            $SubscriptionId = (Get-AzContext).Subscription.Id
+        }
+
+        if ($PSBoundParameters.ContainsKey("CloudService"))
+        {
+            $elements = $CloudService.Id.Split("/")
```
Comment (msJinLei) [inline src/CloudService/custom/Get-AzCloudServicePublicIPAddress.ps1:50]: Please refer to the comments of Get-AzCloudServiceNetworkInterface.ps1

File: src/CloudService/custom/New-AzCloudServiceDiagnosticsExtension.ps1 [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    $privateConfig = $diagnosticsConfigurationXml.ChildNodes | Where-Object { $_.Name -eq 'PrivateConfig' }
+    if ($privateConfig)
+    {
+        $protectedSetting = $privateConfig.OuterXml
+        Write-Host "Using PrivateConfig information defined in diagnostics configuration file."
+    }
+    else
+    {
+        $storageEndpoint = "https://core.windows.net"
+        $protectedSetting = '<?xml version="1.0" encoding="UTF-8"?><PrivateConfig xmlns="http://schemas.microsoft.com/ServiceHosting/2010/10/DiagnosticsConfiguration"><StorageAccount endpoint="' + $storageEndpoint + '" key="' + $StorageAccountKey + '" name="' + $storageAccountName + '"/></PrivateConfig>'
```
Comment (msJinLei) [inline src/CloudService/custom/New-AzCloudServiceDiagnosticsExtension.ps1:112]: It is better to use "${variable}" in the string.

File: src/CloudService/custom/New-AzCloudServiceRemoteDesktopExtensionObject.ps1 [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    [string[]] $RolesAppliedTo,
+
+    [Boolean] $AutoUpgradeMinorVersion
+  )
+
+  process {
+    $RDPPublisher = "Microsoft.Windows.Azure.Extensions"
+    $RDPExtensionType = "RDP"
+
+    $rdpSetting = "<PublicConfig><UserName>" + $Credential.UserName + "</UserName><Expiration>" + $Expiration +"</Expiration></PublicConfig>";
```
Comment (msJinLei) [inline src/CloudService/custom/New-AzCloudServiceRemoteDesktopExtensionObject.ps1:51]: you can use ${variable} in the string

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/CloudService/custom/Switch-AzCloudService.ps1 [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+        }
+
+        # Check that Public IPs counts are correct for source
+        $validSourceIP = ValidateCloudServicePublicIPAddress($SourceCloudService)
+        if ($validSourceIP -eq $false) {
+            throw "Specified source cloud service must have a single public IP address specified in its FrontendIpConfigurations." 
+        }
+  
+        # Parse target cloud service fields
+        $elements = $SourceCloudService.NetworkProfile.SwappableCloudService.Id.Split("/")
```
Comment (msJinLei) [inline src/CloudService/custom/Switch-AzCloudService.ps1:109]: subscript out of bound.

File: src/CloudService/custom/Switch-AzCloudService.ps1 [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+        $requestBody = $requestBody -replace "#LBFE1#", $sourceLB.properties.frontendIPConfigurations[0].Id
+        $requestBody = $requestBody -replace "#PIP2#", $TargetCloudService.NetworkProfile.LoadBalancerConfiguration[0].FrontendIPConfiguration[0].PublicIPAddressId
+        $requestBody = $requestBody -replace "#LBFE2#", $targetLB.properties.frontendIPConfigurations[0].Id
+        $requestBody = $requestBody -replace "#PIP1#", $SourceCloudService.NetworkProfile.LoadBalancerConfiguration[0].FrontendIPConfiguration[0].PublicIPAddressId
+
+        # Set up API URI and Headers
+        $uriToInvoke = "/subscriptions/" + $SubscriptionId + "/providers/Microsoft.Network/locations/" + $SourceCloudService.Location + "/setLoadBalancerFrontendPublicIpAddresses?api-version=" + $ApiVersion
+
+        # Display the information about the VIP swap being made
+        Write-Host "Performing switch cloud service (VIP swap) action between" $SourceCloudService.Name "and" $TargetCloudService.Name
```
Comment (msJinLei) [inline src/CloudService/custom/Switch-AzCloudService.ps1:147]: You can also use multi-line string here.

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please check whether the parameters' descriptions are well generated in cmdlet help files.

File: src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1 [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+        [Parameter(ParameterSetName="CloudServiceName", HelpMessage="RoleInstanceName.")]
+        [Parameter(ParameterSetName = "CloudService", HelpMessage="RoleInstanceName.")]
+        [string] $RoleInstanceName,
+
+        [Parameter(Mandatory=$true, ParameterSetName="CloudService", HelpMessage="CloudService instance.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.CloudService.Models.Api20201001Preview.CloudService] $CloudService,
+
+        [Parameter(DontShow)]
+        [string] $ApiVersion = "2020-06-01"
```
Comment (dolauli) [inline src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1:45]: Who will set the $ApiVersion without using "2020-06-01"

File: src/CloudService/custom/New-AzCloudServiceLoadBalancerConfigurationObject.ps1 [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+.Synopsis
+Create a in-memory object for LoadBalancerConfiguration
+.Description
+Create a in-memory object for LoadBalancerConfiguration
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.CloudService.Models.Api20201001Preview.LoadBalancerConfiguration
+.Link
+https://docs.microsoft.com/en-us/powershell/module/az.CloudService/new-AzCloudServiceLoadBalancerConfigurationObject
+#>
```
Comment (dolauli) [inline src/CloudService/custom/New-AzCloudServiceLoadBalancerConfigurationObject.ps1:26]: I am think if the model is very simple. Just like this one, there is only two members. Shall we still need to create a cmdlet for it

File: src/CloudService/custom/New-AzCloudServiceDiagnosticsExtension.ps1 (hunk 4 above)
Comment (wyunchi-ms) [reply src/CloudService/custom/New-AzCloudServiceDiagnosticsExtension.ps1:113]: There will be many escape char for `"` in xml.

File: src/CloudService/custom/New-AzCloudServiceLoadBalancerConfigurationObject.ps1 (hunk 9 above)
Comment (wyunchi-ms) [reply src/CloudService/custom/New-AzCloudServiceLoadBalancerConfigurationObject.ps1:26]: Service team persist to keep it to provide a powershell way to create the object.

File: src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1 (hunk 8 above)
Comment (wyunchi-ms) [reply src/CloudService/custom/Get-AzCloudServiceNetworkInterface.ps1:50]: Agree

File: src/CloudService/custom/Switch-AzCloudService.ps1 [hunk 10]
```diff
@@ -96,7 +96,7 @@ function Switch-AzCloudService {
 
         # Check that both have swappable property set.
         if ([string]::IsNullOrEmpty($SourceCloudService.NetworkProfile.SwappableCloudService.Id)) {
-            throw "SwappableCloudServiceId is not set on the source cloud service " + $SourceCloudService.Name 
+            throw "SwappableCloudServiceId is not set on the source cloud service ($SourceCloudService.Name)"
```
Comment (msJinLei) [inline src/CloudService/custom/Switch-AzCloudService.ps1:96]: As I tried, variable reference should be $($SourceCloudService.Name). Have you tried it?

File: src/CloudService/custom/Switch-AzCloudService.ps1 [hunk 11]
```diff
@@ -141,16 +141,24 @@ function Switch-AzCloudService {
...
-        Write-Host $requestBody
+        Write-Host "Performing switch cloud service (VIP swap) action between ($SourceCloudService.Name) and ($TargetCloudService.Name)
+
+Request URI : $uriToInvoke
+POST
+
+Request Body :
+$requestBody"
+
+        # Write-Host "Performing switch cloud service (VIP swap) action between ($SourceCloudService.Name) and ($TargetCloudService.Name)"
```
Comment (msJinLei) [inline src/CloudService/custom/Switch-AzCloudService.ps1:156]: remove comment

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**